algorithms/C/searching/Interpolation-search.c

```diff
@@ -0,0 +1,72 @@
+/*
+Program to search data in array using interpolation search
+Interpolation search: Used to search data in a sequence stored array
+*/
+
+#include <stdio.h>
+
+#include <stdlib.h>
+ // Function to search data in a given array
+int interpolation_search(int * arr, int low, int high, int to_search) {
+  if (low <= high && to_search >= arr[low] && to_search <= arr[high]) {
+        if(arr[high]==arr[low]&&to_search==arr[high]&&arr[low]==to_search)
+        return high;
+    // formula to find index
```

Comment:
No need to perform extra comparison as the third condition will be a tautology if the first 2 are true.

```suggestion
  if (low <= high && to_search >= arr[low] && to_search <= arr[high]) {
      if( arr[high] == arr[low] && to_search==arr[high] )
        return high;
    // formula to find index
```